### DISCUSSION TITLE
Is there multi-gpu versions for inference ?

### DISCUSSION THREAD

**Discussion by fgbxx** [MIC-DKFZ/nnUNet#780] [Q&A] [answered]
HI, all 
   For the image volume is very large , for example, it is 800*512*512 pixels, maybe patches number will be 700 or more. so the prediction is very slow for single GPU.
   I just want to know if have multi-GPUs, is there someone that had multi-gpus versions or share some implementation steps or examples?
   very thankful for your feedback.

Best, 
Gang

**Comment by FabianIsensee** [comment, marked as answer]
Hi Gang,
you can use multiple GPUs for inference by parallelizing the inference on a per-data level.
Example:
```bash
CUDA_VISIBLE_DEVICES=0 nnUNet_predict [...] --part_id 0 --num_parts 2
CUDA_VISIBLE_DEVICES=1 nnUNet_predict [...] --part_id 1 --num_parts 2
```
(run these two at the same time)
This will run half of the images on GPU0 and the other half on GPU1.

**Reply by Peaceandmaths** [reply]
Can i do the same for training ? Do I understand correctly, I can define an arbitrary number of parts depending on the volume of my data ?
